src/app/market/trades/tradesBoxPlot/tradesBoxPlot.component.ts

```diff
@@ -0,0 +1,202 @@
+import _ from 'lodash';
+import * as d3 from 'd3';
+import moment from 'moment';
+import {Component, Input, OnChanges, SimpleChanges} from '@angular/core';
+
+@Component({
+  selector: 'app-market-trades-box-plot',
+  templateUrl: 'tradesBoxPlot.component.html'
+})
+export class TradesBoxPlotComponent implements OnChanges {
+  @Input() data: Array<any> = [];
+
+  private config = {
+    width: null,
+    height: null,
+    totalWidth: 900,
+    totalHeight: 600,
+    barWidth: 20,
+    margin: {
+      top: 20,
+      right: 10,
+      bottom: 20,
+      left: 35
+    }
+  };
+  public preparedData;
+  private xScale;
+  private volumeYScale;
+  private boxPlotYScale;
+  private averageLine;
+  public itemToView;
+  public currency;
+
+  constructor() {
+    _.set(this, 'config.width', this.config.totalWidth - this.config.margin.left - this.config.margin.right);
+    _.set(this, 'config.height', this.config.totalHeight - this.config.margin.top - this.config.margin.bottom);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const volumes = _.get(this, 'data.volumes', []);
+    const boxPlot =  _.get(this, 'data.boxPlotChartData', []);
+    this.preparedData = this.parseData(this.mergeData(boxPlot, volumes,'period_start'));
+    this.currency = _.get(this, 'data.currency');
+    this.render();
+  }
+
+  private mergeData = (a1, a2, key) => _.map(a1, i => ({...i, ..._.find(a2, [key, i[key]])}));
+
+  private parseData(d) {
+    return _.chain(d)
+      .sortBy('period_start')
+      .map(value => {
+        for (let key in value) {
```

Comment:
What is this loop doing?

Reply:
Part of the return values is in 'string' this loop changes them to Number if it is possible.

Reply:
It would be safer to convert only known keys.

Reply:
OK